src/term/mod.rs

```diff
@@ -99,6 +107,48 @@ impl Search for Term {
         point
     }
 
+    fn bracket_pair_search(&self, start: Point<usize>, direction: Direction, start_char: char, end_char: char) -> Option<Point<usize>> {
+        let mut iter = self.grid.iter_from(start.into());
+        // For every character match that equals the starting bracket, we
+        // ignore one bracket of the opposite type.
+        let mut skip_pairs = 0;
+
+        let mut match_bracket = |c: char| -> bool {
+            if c == start_char {
+                skip_pairs += 1;
+            }
+
+            if c == end_char {
+                if skip_pairs == 0 {
+                    return true;
+                }
+
+                skip_pairs -= 1;
+            }
+
+            false
+        };
+
+        match direction {
+            Direction::Right => {
+                while let Some(cell) = iter.next() {
+                    if match_bracket(cell.c) {
+                        return Some(iter.cur);
+                    }
+                }
```

Comment:
Instead of using `while` loops on iterators, `for` loops should be preferred in general. However we can go even one step further here.

The following should do the job:
`if let Some(cell) = iter.find(|cell| match_bracket(cell.c)) { return Some(cell) }`